osquery/tables/networking/darwin/wifi_status.mm

```diff
@@ -12,20 +12,51 @@
 
 #include <osquery/core/system.h>
 #include <osquery/core/tables.h>
+#include <osquery/logger/logger.h>
 #include <osquery/tables/networking/darwin/wifi_utils.h>
+#include <osquery/utils/darwin/system_profiler.h>
 
 namespace osquery {
 namespace tables {
 
 QueryData genWifiStatus(QueryContext& context) {
   QueryData results;
   @autoreleasepool {
-    std::string interfaceName = "en0";
+    // Get the list of Wifi interfaces on the system.
     NSArray<CWInterface*>* interfaces =
         [[CWWiFiClient sharedWiFiClient] interfaces];
     if (interfaces == nil || [interfaces count] == 0) {
       return results;
     }
+
+    // Get Airport data from system profiler, in order to get the current
+    // network name for systems that don't return it via CWWiFiClient.
+    NSDictionary* __autoreleasing result;
+    Status status = getSystemProfilerReport("SPAirPortDataType", result);
+    if (!status.ok()) {
+      LOG(ERROR) << "failed to get SPAirPortDataType config: "
+                 << status.getMessage();
+      return {};
+    }
+
+    // Make a map of interface names to network names.
+    NSMutableDictionary* wifiNetworks = [NSMutableDictionary dictionary];
+    for (NSDictionary* item in [result objectForKey:@"_items"]) {
+      // Get the item's airport intefaces, which will usually include at least
+      // a wifi card and an Apple Wireless Direct Link (AWDL) interface.
+      NSArray* airportInterfaces =
+          [item objectForKey:@"spairport_airport_interfaces"];
+      // Get the wifi interface (the one starting with "en").
+      NSDictionary* wifiInterface = [[airportInterfaces
+          filteredArrayUsingPredicate:
+              [NSPredicate predicateWithFormat:@"_name BEGINSWITH %@", @"en"]]
+          lastObject];
+      // Add the network name to the map, indexed by interface name.
+      wifiNetworks[[wifiInterface objectForKey:@"_name"]] =
+          [[wifiInterface objectForKey:@"spairport_current_network_information"]
+              objectForKey:@"_name"];
+    }
```

Comment:
Re-worked this to build the map based on interface name (e.g. `en0`) instead of MAC address.  When testing this on a newer Macbook, it had the "Private WiFi Address" feature turned on which made the address in the system profiler not match the address from `CWWiFiClient`, which in turn caused a fault when trying to read from the map below.